src/browser/api/system.js

```diff
@@ -566,12 +566,15 @@ module.exports.System = {
             downloadId: downloadId
         };
 
-        const publishSuccess = rvmBus.publish(rvmMessage);
-
-        if (publishSuccess) {
-            cb(null, downloadId);
+        const publishSuccess = rvmBus.publish(rvmMessage, response => {
+            if (response.error) {
+                cb(new Error(response.error));
+            } else {
+                cb();
```

Comment:
That was the old style, as of [This PR](https://github.com/openfin/javascript-adapter/pull/303) the download ID's are generated by the renderer so no need to provide them.